microsoft-edge/extensions-chromium/publish/api/using-addons-publish-api.md

```diff
@@ -0,0 +1,222 @@
+
+The tentative date for the availability of the APIs is December 2021.
+
+
+<!-- ====================================================================== -->
+## Terminology
+
+| Term | Definition |
+|---|---|
+| _product_ | A single product for which you have an entire entry at Partner Center. |
+| _product ID_ | The product ID of the product whose draft needs to be published.  This is the product ID of your product as shown at Partner Center. |
+| _submission_ | A product which has not yet been approved for publishing at Partner Center.  The process of submitting a product for approval has several steps. |
+| _operation_ | A REST operation such as GET or PUT.  There are several operations, defined at [Add-ons Publish API Reference][AddonsPublishAPIRef]. |
+| _operation ID_ | The ID of a REST operation. |
+
+
+<!-- ====================================================================== -->
+## Before you begin
+
+To use the Microsoft Edge Add-ons Publish API, you need to enable the API for your project in the Microsoft Partner Center.
+
+1. Visit Microsoft Partner Center and login to the account from which you have already published an add-on.
+
+1. Under the **Microsoft Edge** program, a new page for **Publish APIs** appears.
+
+1. Enable the APIs.<!-- what UI action to do that?--> This step may take a few minutes.  After the APIs are enabled, the **Client ID**, **Client Secret**, and **Auth Token URL** are displayed on this page.
+
+1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**. You will use them in the next step to get an access token.
+
+
+<!-- ====================================================================== -->
+## Retrieving the access token
+
+After you've acquired the necessary authorization for your application, proceed with acquiring access tokens for APIs.
+
+1. To get a token using the client credentials grant, send a POST request to the Auth Token URL.  The tenant information is available in the URL that you received in the **Before you begin** steps above.
+
+```console
+Endpoint: https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token
+Type: POST
+Header Parameters: Content-Type: application/x-www-form-urlencoded
+```
+
+### Sample request
+
+```console
+> curl \
+-X POST \
+-H "Content-Type: application/x-www-form-urlencoded" \
+-d "client_id={$Client_ID}" \
+-d "scope=https://addons.edge.microsoft.com/.default" \
+-d "client_secret={$Client_Secret}" \
+-d "grant_type=client_credentials" \
+-v \
+https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token
+```
+
+### Sample response
+
+```json
+{
+  "token_type": "Bearer",
+  "expires_in": 3599,
+  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1uQ19WWmNBVGZNNXBP..."
+}
+```
+
+For more information, navigate to [OAuth 2.0 client credentials flow on the Microsoft identity platform][AzureOAuthGetToken].
+
+
+<!-- ====================================================================== -->
+## Using the Publish API
```

Comment:
Dissolve this section by merging each subsection into the corresponding subsection of the API Ref article?  The current design might be usable, b/c it has consistent cross-links between the corresponding subsections of the two articles.

Reply:
Not sure about the change that is proposed in this? Could you clarify if there is a question for me.

Reply:
I'm wondering why a given REST endpoint has one subsection in one article, and another, parallel subsection in a separate (API Ref) article.  I'm wondering if the two sections for a given endpoint should be placed consecutively in the Ref article since both sections look like Ref content.  Elsewhere, I realize, some Ref articles became over-filled with verbose usage info, so we want to keep Ref content streamlined (and cross-link).